src/combat.cpp

```diff
@@ -1175,197 +1225,99 @@ void TargetCallback::onTargetCombat(Creature* creature, Creature* target) const
 
 //**********************************************************//
 
-void AreaCombat::clear()
-{
-	for (const auto& it : areas) {
-		delete it.second;
-	}
-	areas.clear();
+MatrixArea MatrixArea::flip() const {
+    Container newArr(arr.size());
+    for (auto i = 0; i < rows; ++i) {
```

Comment:
it is best to use the specific numeric type to avoid the `C4018` warning in compiler output.

` for (uint32_t i = 0; i < rows; ++i) {`